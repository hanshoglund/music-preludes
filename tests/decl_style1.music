
data Foo = Foo | Bar

scale Foo = scat [c,d,e,f,g,a,g,f]|/8
scale Bar = scale Foo

triad a = a <> up _M3 a <> up _P5 a

example = up _P8 (scale Foo) </> (triad c)|/2 |> (triad g_)|/2

