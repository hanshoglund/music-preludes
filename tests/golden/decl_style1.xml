<?xml version='1.0' ?>
<score-partwise>
  <movement-title></movement-title>
  <identification>
    <creator type="composer"></creator>
  </identification>
  <part-list>
    <score-part id="P1">
      <part-name></part-name>
      <part-name-display>
        <display-text></display-text>
      </part-name-display>
    </score-part>
    <score-part id="P2">
      <part-name></part-name>
      <part-name-display>
        <display-text></display-text>
      </part-name-display>
    </score-part>
  </part-list>
  <part id="P1">
    <measure number="1">
      <attributes>
        <divisions>768</divisions>
      </attributes>
      <attributes>
        <key>
          <fifths>0</fifths>
          <mode>major</mode>
        </key>
      </attributes>
      <direction>
        <direction-type>
          <metronome>
            <beat-unit>quarter</beat-unit>
            <per-minute>120</per-minute>
          </metronome>
        </direction-type>
      </direction>
      <attributes>
        <staves>1</staves>
      </attributes>
      <attributes>
        <clef>
          <sign>G</sign>
          <line>2</line>
        </clef>
      </attributes>
      <attributes>
        <time>
          <beats>4</beats>
          <beat-type>4</beat-type>
        </time>
      </attributes>
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>C</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <pitch>
          <step>D</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <pitch>
          <step>E</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <pitch>
          <step>F</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <pitch>
          <step>G</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <pitch>
          <step>A</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <pitch>
          <step>G</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <pitch>
          <step>F</step>
          <alter>0.0</alter>
          <octave>5</octave>
        </pitch>
        <duration>384</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
    </measure>
  </part>
  <part id="P2">
    <measure number="1">
      <attributes>
        <divisions>768</divisions>
      </attributes>
      <attributes>
        <key>
          <fifths>0</fifths>
          <mode>major</mode>
        </key>
      </attributes>
      <direction>
        <direction-type>
          <metronome>
            <beat-unit>quarter</beat-unit>
            <per-minute>120</per-minute>
          </metronome>
        </direction-type>
      </direction>
      <attributes>
        <staves>1</staves>
      </attributes>
      <attributes>
        <clef>
          <sign>G</sign>
          <line>2</line>
        </clef>
      </attributes>
      <attributes>
        <time>
          <beats>4</beats>
          <beat-type>4</beat-type>
        </time>
      </attributes>
      <direction>
        <direction-type>
          <dynamics>
            <mf />
          </dynamics>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>C</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>1536</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
      <note>
        <chord />
        <pitch>
          <step>E</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>1536</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
      <note>
        <chord />
        <pitch>
          <step>G</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>1536</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
      <note>
        <pitch>
          <step>G</step>
          <alter>0.0</alter>
          <octave>3</octave>
        </pitch>
        <duration>1536</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
      <note>
        <chord />
        <pitch>
          <step>B</step>
          <alter>0.0</alter>
          <octave>3</octave>
        </pitch>
        <duration>1536</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
      <note>
        <chord />
        <pitch>
          <step>D</step>
          <alter>0.0</alter>
          <octave>4</octave>
        </pitch>
        <duration>1536</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
    </measure>
  </part>
</score-partwise>
